massiv/src/Data/Massiv/Core/Index/Ix.hs

```diff
@@ -81,6 +85,9 @@ pattern Ix5 i j k l m = i :> j :> k :> l :. m
 -- | n-dimensional index. Needs a base case, which is the `Ix2`.
 data IxN (n :: Nat) = (:>) {-# UNPACK #-} !Int !(Ix (n - 1))
 
+instance Validity (Ix (n - 1)) => Validity (IxN n) where
+  validate (_ :> b) = delve "tail" b
```

Comment:
Is there a reason that the head doesn't need to be validated?

Reply:
The head is an Int, which is always valid.
I could make this "trivialValidation" though.
The reason I delved into the tail, is for maintainability:
If you put a restriction on Ix0, Ix1, Ix2 later on, this would cause unexpected behavior
in higher dimensional index types.